CMakeLists.txt

```diff
@@ -311,30 +312,8 @@ endif()
 set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIRS})
 set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${ARMADILLO_LIBRARIES})
 
-# Find stb_image.h and stb_image_write.h.
-if (NOT DOWNLOAD_DEPENDENCIES)
-  find_package(StbImage)
-else()
-  find_package(StbImage)
-  if (NOT StbImage_FOUND)
-    get_deps(https://mlpack.org/files/stb.tar.gz stb stb.tar.gz)
-    set(STB_IMAGE_INCLUDE_DIR ${GENERIC_INCLUDE_DIR})
-    find_package(StbImage REQUIRED)
-  endif()
-endif()
-
-if (StbImage_FOUND)
-  set(STB_AVAILABLE "1")
-  set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} "${STB_IMAGE_INCLUDE_DIR}")
-
-  # Make sure that we can link STB in multiple translation units.
-  include(CMake/TestStaticSTB.cmake)
-  if (NOT CMAKE_HAS_WORKING_STATIC_STB)
-    message(FATAL_ERROR "STB implementations's static mode cannot link across "
-        "multiple translation units!  Try upgrading your STB implementation, "
-        "or using the auto-downloader (set DOWNLOAD_DEPENDENCIES=ON in the "
-        "CMake configuration command.")
-  endif ()
+if (DONT_USE_MLPACK_STB)
+  find_package(StbImage REQUIRED)
```

Comment:
Ok, so no more autodownloading of STB (that's fine with me, I just want to check that that's what you intended).

I think we should keep the `CMAKE_HAS_WORKING_STATIC_STB` check though: if the user wants to use a version of STB that is too old (like whatever ships on Debian stable right now), it will fail the check.

Reply:
Agree this would make sense to have it, even the relevant code for it I did not remove